hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/WritableKVStateBase.java

```diff
@@ -170,6 +174,17 @@ public final Set<K> modifiedKeys() {
         return modifications.keySet();
     }
 
+    /** {@inheritDoc}
+     * To add the comparator to the underlying data source, we create a new TreeMap with the Comparator
+     * This should be called on each writable store constructor until all other types of Key Comparator are implemented.
+     * */
+    @Override
+    public final void updateComparator(@NonNull final Comparator<K> comparator) {
```

Comment:
I think the `Comparator` should be set during construction and never changed again.

Reply:
Yes. Each store is responsible to update their comparator.

Reply:
Sorry for not being clear. I would have expected that the `Comparator` is immutable. There is probably no valid use case where we want to change the `Comparator` later, or is there? If the `Comparator` is immutable, the code should reflect that.

Reply:
This method has been removed. WritableTopicStore uses a private static final Comparator.